test/surrogate.test.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const test = require('tap').test
+const validator = require('is-my-json-valid')
+const build = require('..')
+
+test(`render a string with surrogate pairs as JSON:test 1`, (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'surrogate',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('𝌆')
+
+  t.equal(output, '"𝌆"')
+  t.ok(validate(JSON.parse(output)), 'valid schema')
+})
+
+test(`render a string with surrogate pairs as JSON: test 2`, (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'long',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('\uD834\uDF06')
+
+  t.equal(output, '"𝌆"')
+  t.ok(validate(JSON.parse(output)), 'valid schema')
+})
+
+test(`render a string with Unpaired surrogate code as JSON`, (t) => {
+  t.plan(1)
+
+  const schema = {
+    title: 'surrogate',
+    type: 'string'
+  }
+
+  const validate = validator(schema)
+  const stringify = build(schema)
+  const output = stringify('\uDF06\uD834')
+  // t.equal(output, '"\\udf06\\ud834"')
```

Comment:
It would be great to use a mirror test for this.

```js
t.equal(output, JSON.stringify('\uDF06\uD834'))
```

Reply:
What do you mean by mirror test?(Excuse me if this question is simple :)) )

Reply:
What I put above. Basically you have a target system A. You want to verify that B calls A and returns that output. You call B() and then A() and you compare results. In this way any change of output in A() will be automatically reflected in B() output.

Reply:
Thanks. I got it.